stdlib/http/http.go

```diff
@@ -0,0 +1,235 @@
+package http
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/iocounter"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/log"
+)
+
+// maxResponseBody is the maximum response body we will read before just discarding
+// the rest. This allows sockets to be reused.
+const maxResponseBody = 512 * 1024 // 512 KB
+
+var httpGetStatusCodeOnly = values.NewFunction(
+	"get",
+	semantic.NewFunctionPolyType(semantic.FunctionPolySignature{
+		Parameters: map[string]semantic.PolyType{
+			"url":     semantic.String,
+			"headers": semantic.Tvar(1),
+			"data":    semantic.Bytes,
+		},
+		Required: []string{"url"},
+		Return:   semantic.Int,
+	}),
+	func(ctx context.Context, args values.Object) (values.Value, error) {
+		// Get and validate URL
+		uV, ok := args.Get("url")
+		if !ok {
+			return nil, errors.New(codes.Invalid, "missing \"url\" parameter")
+		}
+		u, err := url.Parse(uV.Str())
+		if err != nil {
+			return nil, err
+		}
+		deps := flux.GetDependencies(ctx)
+		validator, err := deps.URLValidator()
+		if err != nil {
+			return nil, err
+		}
+		if err := validator.Validate(u); err != nil {
+			return nil, err
+		}
+
+		// Construct data
+		// Yes, you can send a request body with GET but it should not have any meaning.
+		// If you give it meaning by parsing it on the server and changing your response based on its contents,
+		// then you are ignoring this recommendation.
+		var data []byte
+		dataV, ok := args.Get("data")
+		if ok {
+			data = dataV.Bytes()
+		}
+
+		// Construct HTTP request
+		req, err := http.NewRequest("GET", uV.Str(), bytes.NewReader(data))
+		if err != nil {
+			return nil, err
+		}
+
+		// Add headers to request
+		header, ok := args.Get("headers")
+		if ok && !header.IsNull() {
+			var rangeErr error
+			header.Object().Range(func(k string, v values.Value) {
+				if v.Type() == semantic.String {
+					req.Header.Set(k, v.Str())
+				} else {
+					rangeErr = errors.Newf(codes.Invalid, "header value %q must be a string", k)
+				}
+			})
+			if rangeErr != nil {
+				return nil, rangeErr
+			}
+		}
+
+		// Perform request
+		dc, err := deps.HTTPClient()
+		if err != nil {
+			return nil, errors.Wrap(err, codes.Aborted, "missing client in http.get")
+		}
+
+		statusCode, err := func(req *http.Request) (int, error) {
+			s, cctx := opentracing.StartSpanFromContext(ctx, "http.get")
+			s.SetTag("url", req.URL.String())
+			defer s.Finish()
+
+			req = req.WithContext(cctx)
+			response, err := dc.Do(req)
+			if err != nil {
+				return 0, err
+			}
+
+			// Read the response body but limit how much we will read.
+			// This is to allow a socket to be reused after it is closed.
```

Comment:
I am not familiar with this idea of a limit reader that enables the socket to be reused. Can you point to some docs somewhere that explain this?

Reply:
https://github.com/influxdata/flux/blob/7ad7e2ba275c45e21a5109766b36c8848e9a4e0f/stdlib/http/post.go#L104

Reply:
I see it comes from e35a585d1. Basically if we can read the response body then we can reuse the connection. If we don't read it then it just need to be closed to tell the other side we don't want the rest of the data. But in that case we can't reuse the connection. But instead of just reading all of the response we only read up to 512KB to capture most use cases while still protecting us against very large responses.